fenix/app/src/main/java/org/mozilla/fenix/home/HomeScreenViewModel.kt

```diff
@@ -11,4 +11,10 @@ class HomeScreenViewModel : ViewModel() {
      * Used to delete a specific session once the home screen is resumed
      */
     var sessionToDelete: String? = null
+
+    /**
+     * Used to track if the last remaining tab closed was private. Only used by TabStrip to show
+     * undo snackbar on Home when navigating back from BrowserFragment to Home.
+     */
+    var wasLastTabsClosedPrivate: Boolean? = null
```

Comment:
Let's not add to this if possible. We should probably prefer our Stores.

Reply:
What led to needing this workaround? We're able to show the undo snackbar when closing the last remaining private tab via the Tabs Tray.

see: the usage of `dismissTabsTrayAndNavigateHome(...)` within [`TabsTrayController.deleteTab()`](
https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt#301-325)

Reply:
So this was needed as we navigate from `BrowserFragment` to `HomeFragment` when the last tab is closed, just passing the callback to show snackbar in BrowserFragment doesn't work for this last tab case. Tabs tray implementation also does something similar using `homeScreenViewModel.sessionToDelete` for the last tab case. In the link, `dismissTabsTrayAndNavigateHome` calls `navigateToHomeAndDeleteSession` which is a function  defined in `TabsTrayFragment`. [source](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt#707). This sets `sessionToDelete` that's checked in `HomeFragment.onViewCreated` to show the snackbar and remove the tab.

❓ Or am i misunderstanding this?

💡 Could also simply do something similar but then the removal of tab for this case (last tab) will be handled differently than the rest of tabs, as the `sessionToDelete` logic in `HomeFragment` finds and removes the tab and then goes on to show the snackbar. 

🏗️ Ideally, I think we need something like a `SharedFlow` here for displaying snackbars which is good for nav events that can be set similar to set on dispatched actions. Simply observing state using the current abstractions will collect every time i.e show snackbar every time the screen collects the state. 

👂 I'm all ears for any ideas here.

Reply:
I've update to use the AppStore. Note that this cannot be observed using the existing abstractions for the same reasons as above so we indeed need to do a check and then set it to null, similar to what was happening with the VM. LMKWYT!